common/spec/dependabot/pull_request_creator/message_builder/link_and_mention_sanitizer_spec.rb

```diff
@@ -68,11 +69,22 @@
 
       context "that appears in codeblock quotes" do
         let(:text) { "``` @model ||= 123```" }
-        it { is_expected.to eq(text) }
+        it do
+          is_expected.to eq(
+            "<pre><code></code></pre>\n<p>"\
+            "<a href=\"https://github.com/model\">@​model</a> ||= 123```</p>\n"
```

Comment:
@rebelagentm ohh it looks like we need to fix the `@mention` sanitizer in this PR because this is now parsing part of the string as markdown and converting it to html instead of the whole thing (I'm pretty sure the mention link should be inside the `<pre><code></code></pre>` tags)

A good starting point would be gutting most of `sanitize_links_and_mentions` and setting up   `CommonMarker.render_doc..` there and then passing around `doc` instead of `text`.

Reply:
Or actually, you might be able to do this later if you remove `line = sanitize_links(line)` from the loop and call it on the entire text at the end of sanitize_links_and_mentions, like: `sanitize_links(lines.join)`

Reply:
Cool.  I was just trying to figure out what was going on with those tests and `mention_sanitizer`.  Thanks for pointing me in the right direction!

Reply:
I think these URLs should be swapped out to the redirect one, this might get fixed with the above change